sympy/calculus/singularities.py

```diff
@@ -50,13 +51,18 @@ def singularities(expression, symbol):
     .. [1] http://en.wikipedia.org/wiki/Mathematical_singularity
 
     """
+    from sympy.sets import FiniteSet
+
     if not expression.is_rational_function(symbol):
         raise NotImplementedError(
             "Algorithms finding singularities for non-rational"
             " functions are not yet implemented."
         )
     else:
-        return solveset(simplify(1 / expression), symbol)
+        singularities = []
+        for d in denoms(expression):
+            singularities.extend(solveset(d, symbol))
+        return FiniteSet(*singularities)
```

Comment:
@smichr I think that we should make a note of the fact that this function returns only the isolated singularities of the given expression. ie. we need to state that non-isolated singularities and branch points are not determined by this function.